### PR TITLE
Feature/user authentication

### PR DESCRIPTION
**Pull request by Boyok1337**
implement MVC for user
register admin for user model
added JWT auth
update requirements

### CODE REVIEW

**Review by Gilgumish — APPROVED** [review]
well done!

File: library_service/settings.py [hunk 1]
```diff
@@ -37,6 +37,12 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+
+    # External apps
+    'rest_framework_simplejwt',
+    'rest_framework',
+
+    # User apps
```
Comment (VitalyBashkiser) [inline library_service/settings.py:45]: Good idea with the # separation of libraries and apps 👍

File: user/models.py [hunk 2]
```diff
@@ -1,5 +1,47 @@
...
+    first_name = models.CharField(_("first name"), max_length=50)
+    last_name = models.CharField(_("last name"), max_length=50)
+
+    USERNAME_FIELD = "email"
+    REQUIRED_FIELDS = [
+        "first_name",
+        "last_name",
+    ]
+
+    objects = UserManager()
```
Comment (VitalyBashkiser) [inline user/models.py:47]: GJ 👍

File: user/serializers.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        password = validated_data.pop("password", None)
+        user = super().update(instance, validated_data)
+        if password:
+            user.set_password(password)
+            user.save()
+
+        return user
```
Comment (VitalyBashkiser) [inline user/serializers.py:32]: Well done 👍

**Review by VitalyBashkiser — APPROVED** [review]
It looks awesome 👍

File: requirements.txt [hunk 4]
```diff
@@ -1,10 +1,27 @@
...
 Django==4.0.4
 django-debug-toolbar==3.4.0
 djangorestframework==3.13.1
 djangorestframework-simplejwt==5.2.0
 drf-spectacular==0.22.1
-Pillow==10.3.0
 flake8==5.0.4
 flake8-quotes==3.3.1
 flake8-variables-names==0.0.5
+inflection==0.5.1
```
Comment (oleksandra-shershen) [inline requirements.txt:11]: Do we really need all these packages now?

File: user/models.py [hunk 5]
```diff
@@ -1,5 +1,47 @@
...
+        if extra_fields.get("is_superuser") is not True:
+            raise ValueError("Superuser must have is_superuser=True.")
+
+        return self._create_user(email, password, **extra_fields)
 
 
 class User(AbstractUser):
-    pass
+    username = None
+    email = models.EmailField(_("email address"), unique=True)
```
Comment (oleksandra-shershen) [inline user/models.py:37]: (Optional) You can write validators for fields. For example:
```
email = models.EmailField(
        _("email address"),
        unique=True,
        validators=[RegexValidator(
            regex='^[\w\.-]+@[\w\.-]+\.\w+$',
            message=_("Enter a valid email address"),
            code='invalid_email'
        )]
    )
```

File: user/models.py [hunk 6]
```diff
@@ -1,5 +1,47 @@
-from django.contrib.auth.models import AbstractUser
+from django.db import models
+from django.utils.translation import gettext as _
+from django.contrib.auth.models import AbstractUser, BaseUserManager
+
+
+class UserManager(BaseUserManager):
+    use_in_migrations = True
+
+    def _create_user(self, email, password, **extra_fields):
```
Comment (oleksandra-shershen) [inline user/models.py:9]: (Optional) Added docstrings to classes and methods to describe their purpose

**Review by oleksandra-shershen — CHANGES_REQUESTED** [review]

**Review by oleksandra-shershen — APPROVED** [review]

**Pull request merged**